.circleci/config.yml.in

```diff
@@ -87,10 +87,62 @@ jobs:
     steps:
       - checkout
       - run:
+          name: Install check utilities
+          command: pip install --user --progress-bar=off jinja2 pyyaml
+      - run:
+          name: Check CircleCI config consistency
+          command: python .circleci/regenerate.py && git diff --quiet
+      - run:
+          when: on_fail
+          name: .circleci/config.yml not in sync with config.yml.in! Run '$ python .circleci/regenerate.py' to fix this.
+          command: exit 1
+
+  lint_python_and_config:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - checkout
+      - run:
+          name: Install lint utilities
+          command: pip install --user --progress-bar=off pre-commit
+      - run:
+          name: Install pre-commit hooks
+          command: pre-commit install-hooks
+      - run:
+          name: Lint Python code and config files
+          command: pre-commit run --all-files
+      - run:
+          when: on_fail
+          name: Code format not compliant with the rules! Run '$ pre-commit run --all-files' to fix this.
+          command: exit 1
+
+  lint_c:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - run:
+          name: Install additional system libraries
+          command: |
+            sudo apt update -qy
+            sudo apt install libtinfo5
+      - checkout
+      - run:
+          name: Install lint utilities
           command: |
-            pip install --user --progress-bar off jinja2 pyyaml
-            python .circleci/regenerate.py
-            git diff --exit-code || (echo ".circleci/config.yml not in sync with config.yml.in! Run .circleci/regenerate.py to update config"; exit 1)
+            curl https://oss-clang-format.s3.us-east-2.amazonaws.com/linux64/clang-format-linux64 -o clang-format
+            chmod +x clang-format
+            ./clang-format --version
+      - run:
+          name: Lint C code
+          command: >
+            python run-clang-format.py
+            --recursive
+            --clang-format-executable=./clang-format
+            torchtext/csrc
```

Comment:
The invocation currently in use does not work. See for example [this CI run](https://app.circleci.com/pipelines/github/pytorch/text/4640/workflows/9ab7a299-555e-45ec-be72-802f5f8412e1/jobs/155606) on `main`:

> .circleci/unittest/linux/scripts/run_style_checks.sh: line 25: git-clang-format: command not found

 This is the same script we use at `torchvision`.

Reply:
Thanks for fixing this. Nice to have the scrip consistent with vision and other domains!

Reply:
I've opted to go for the `pre-commit` hook of `clang-format`. This currently diverges from `torchvision`, but I'll fix this in pytorch/vision#5317.

Reply:
Is the result of clang-format consistent across different OSs?
We fix clang-format by using the same binaries shared on S3.
https://github.com/pytorch/vision/issues/2372

Reply:
I've checked in pytorch/vision#5317 and the binary used by the hook indeed produces a different format than the one PyTorch provides. Thus, I would remove the hook once again and use the same script as `torchvision` does. cc @abhinavarora